app/Http/Controllers/GroupController.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Judite\Models\Group;
+use App\Judite\Models\Course;
+use App\Judite\Models\Student;
+use App\Judite\Models\Invitation;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Auth;
+use App\Exceptions\UserHasAlreadyGroupInCourseException;
+
+class GroupController extends Controller
+{
+    /**
+     * Create a new controller instance.
+     */
+    public function __construct()
+    {
+        $this->middleware('auth');
+        $this->middleware('can.student');
+        $this->middleware('student.verified');
+        $this->middleware('can.group');
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\View\View
+     */
+    public function index()
+    {
+        $enrollments = DB::transaction(function () {
+            $student = Auth::student();
+
+            $enrollments = $student
+                ->enrollments()
+                ->orderByCourse()
+                ->get();
+
+            $this->addStudentGroupInfoToEachEnrollment($student, $enrollments);
+
+            return $enrollments;
+        });
+
+        return view('groups.index', compact('enrollments'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param $courseId
+     *
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function store($courseId)
+    {
+        $group = DB::transaction(function () use ($courseId) {
+            $course = Course::findOrFail($courseId);
+
+            $group = new Group();
+            $group->course_id = $courseId;
+            $group->save();
+
+            return $group;
+        });
+
+        if ($group == null) {
+            return redirect()->back();
+        }
+
+        try {
+            Auth::student()->join($group);
+
+            flash('You have successfully joined a group.')->success();
+        } catch (UserHasAlreadyGroupInCourseException $e) {
+            flash('You already have a group.')->error();
+            $group->delete();
```

Comment:
Creating a record and delete it? 2 queries? Hum...
You should check for the existing a group in this course which this user as membership. In this way, this exception is no longer need because the validation is done before...
But what do you think? Is this work worthy?